datafusion/expr/src/type_coercion/binary.rs

```diff
@@ -623,6 +640,17 @@ fn null_coercion(lhs_type: &DataType, rhs_type: &DataType) -> Option<DataType> {
     }
 }
 
+fn check_time_unit(datatype: DataType) -> Option<TimeUnit> {
```

Comment:
**nitpick:** `check_time_unit` is vague and could be anything. If we are to open a PR to `datafusion`, I think this should be documented.

Reply:
I agree. I changed it by implemented Alex's suggestion and implemented a DocString, please let me know what you think.